data_service/src/match_processing/match_parser.rs

```diff
@@ -1,31 +1,151 @@
+    let mapped_frames = match_frames
+        .iter()
+        .map(|frame| {
+            // Weird that participant_frames is a struct of ( x1, x2,...x10) instead of a vector ( something iteratable) so we convert it to a hashmap ( only so that we can use .map())
+            let participants_frame_as_hashmap: HashMap<
+                String,
+                MatchTimelineInfoFrameParticipantFrame,
+            > = serde_json::from_value(
+                serde_json::to_value(frame.participant_frames.clone()).unwrap(),
+            )
+            .unwrap();
+
+            // Mapping Participant Frames to Champion Frames
+            let champion_mappings: Vec<ChampionFrame> = participants_frame_as_hashmap
+                .iter()
+                .map(|(player_id, player)| {
+                    // Current Player and Opposing Player
+                    let current_player = participant_id_object_map
+                        .get(&player_id.parse::<i32>().unwrap())
+                        .unwrap();
+                    let opposing_player = participant_id_object_map
+                        .get(
+                            &participant_id_opponent_id_map
+                                .get(&player_id.parse::<i32>().unwrap())
+                                .unwrap(),
+                        )
+                        .unwrap();
+
+                    // Finally creating the Champion Frame required by the API
+                    ChampionFrame {
+                        champion_name: current_player.champion_name.clone(),
+                        opposing_champion_name: opposing_player.champion_name.clone(),
+                        position: calculate_position(current_player.individual_position.as_str()),
+                        gold: player.total_gold.to_string(),
+                    }
+                })
+                .collect();
+
+            MatchTimelineFrame {
+                mappings: champion_mappings,
+                frame_time: TimeDelta::seconds(frame.timestamp as i64),
+            }
+        })
+        .collect();
+
+    crate::types::MatchTimeline {
+        frames: mapped_frames,
+        match_id: match_timeline.metadata.match_id.clone(),
+        start_time: DateTime::from_timestamp_millis(
+            match_frames[0].events[0].real_timestamp.unwrap(),
+        )
+        .expect("invalid timestamp")
+        .naive_utc(),
+    }
+}
+
+pub async fn get_puuid_to_champion_mapping(match_data: Match) -> Vec<PuuidToChampionMapping> {
     match_data
         .info
         .participants
         .iter()
         .map(|participant| PuuidToChampionMapping {
             puuid: participant.puuid.clone(),
             champion_name: participant.champion_name.clone(),
+            position: calculate_position(participant.individual_position.as_str()),
         })
         .collect()
 }
 
+fn get_opponent_player_id(current_player_id: i32, match_data: Match) -> i32 {
```

Comment:
again, this information is present in the get_puuid_to_champion_mapping, so we're duplicating some logic here from that function

Reply:
the id mentioned here isn't puuid. my bad if it wasn't clear from the variable names